src/slurm_plugin/slurm_resources.py

```diff
@@ -231,6 +229,38 @@ def is_poweing_up_idle(self):
     def is_ice(self):
         return self.error_code in self.EC2_ICE_ERROR_CODES
 
+    def is_reboot_requested(self):
+        return self.SLURM_SCONTROL_REBOOT_REQUESTED_STATE in self.states
+
+    def is_reboot_issued(self):
+        return self.SLURM_SCONTROL_REBOOT_ISSUED_STATE in self.states
+
+    def is_rebooting(self):
+        """
+        Check if the node is rebooting via scontrol reboot.
+
+        Check that the node is in a state consistent with the scontrol reboot request, which could trigger
+        the health checks.
+        """
+        cond = False
+        if self._is_drain():
+            if self.is_reboot_issued() or self.is_reboot_requested():
+                logger.debug(
+                    "Node state check: node %s in DRAINED but is currently rebooting, ignoring, node state: %s",
+                    self,
+                    self.state_string,
+                )
+                cond = True
+        elif self.is_down():
```

Comment:
`is_down` here doesn't consider nodes in `DOWN+POWER_DOWN` state. So if nodes is in `REBOOT_ISSUED+DOWN+POWER_DOWN` state, node is not considered as rebooting node. It will be treated as unhealthy node and be terminated.
Does this case happen?

Reply:
This case shouldn't happen normally during the `scontrol reboot` workflow, so it should be considered unhealthy.